addons/mail/models/mail_activity.py

```diff
@@ -249,11 +265,23 @@ def action_feedback(self, feedback=False):
                 subtype_id=self.env.ref('mail.mt_activities').id,
                 mail_activity_type_id=activity.activity_type_id.id,
             )
-            message |= record.message_ids[0]
 
-            # Removing attachments res_id to prevent unlinking it
-            # We still want them in the message
-            self.env['ir.attachment'].search([('res_model', '=', self._name), ('res_id', 'in', self.ids)]).write({'res_id': False})
+            # Moving the attachments in the message
+            # TODO: Fix void res_id on attachment when you create
+            # an activity with an image directly see (/web_editor/attachment/add
+            activity_message = record.message_ids[0]
+            message_attachments = self.env['ir.attachment'].browse(activity_attachments[activity.id])
+            if message_attachments:
+                message_attachments.write({
+                    'res_id': activity_message.id,
+                    'res_model': activity_message._name,
+                })
+                activity_message.attachment_ids = message_attachments
```

Comment:
no need to write on the One2many since you wrote on the Many2one just above.

Reply:
I also had this thinking, but these are different

for example for a account invoice report sent by mail, res_model will be account.invoice and it will be linked to a mail.message through another relation:

```py
attachment_ids = fields.Many2many(
        'ir.attachment', 'message_attachment_rel',
        'message_id', 'attachment_id',
        string='Attachments',
        help='Attachments are linked to a document through model / res_id and to the message '
             'through this field.')
```

Reply:
indeed